src/main/java/org/generationcp/middleware/api/germplasmlist/GermplasmListServiceImpl.java

```diff
@@ -358,6 +359,66 @@ public void deleteGermplasmListFolder(final Integer folderId) {
 		return this.daoFactory.getGermplasmListDAO().getGermplasmListDtos(gid);
 	}
 
+	@Override
+	public void performGermplasmListEntriesDeletion(final List<Integer> gids) {
+		final List<Integer> germplasmListIds = this.daoFactory.getGermplasmListDAO().getListIdsByGIDs(gids);
+		if(org.apache.commons.collections.CollectionUtils.isNotEmpty(germplasmListIds)) {
+			final Map<Integer, List<GermplasmListData>> germplasmListDataMap = this.daoFactory.getGermplasmListDataDAO()
+				.getGermplasmDataListMapByListIds(germplasmListIds);
+			final List<GermplasmListData> germplasmListDataToBeDeleted = new ArrayList<>();
+			final List<GermplasmListData> germplasmListDataToBeUpdated = new ArrayList<>();
+			for (final Integer listId : germplasmListIds) {
+				final Iterator<GermplasmListData> iterator = germplasmListDataMap.get(listId).iterator();
+				while (iterator.hasNext()) {
+					final GermplasmListData germplasmListData = iterator.next();
+					if (germplasmListData.getGermplasm() != null && gids.contains(germplasmListData.getGermplasm().getGid())) {
+						iterator.remove();
+						germplasmListDataToBeDeleted.add(germplasmListData);
+					}
+				}
+
+				// Change entry IDs on listData
+				final List<GermplasmListData> listData = germplasmListDataMap.get(listId);
+				Integer entryId = 1;
+				for (final GermplasmListData germplasmListData : listData) {
+					germplasmListData.setEntryId(entryId);
+					entryId++;
+				}
+				germplasmListDataToBeUpdated.addAll(listData);
+			}
+
+			this.deleteGermplasmListData(germplasmListDataToBeDeleted);
+			this.updateGermplasmListData(germplasmListDataToBeUpdated);
+		}
+	}
+
+	private void updateGermplasmListData(final List<GermplasmListData> germplasmListData) {
+		try {
+			for (final GermplasmListData data : germplasmListData) {
+				this.daoFactory.getGermplasmListDataDAO().saveOrUpdate(data);
+			}
+		} catch (final Exception e) {
+
+			throw new MiddlewareQueryException(
+				"Error encountered while saving Germplasm List Data: GermplasmServiceImpl.updateGermplasmListData(germplasmListData="
+					+ germplasmListData + "): " + e.getMessage(),
+				e);
+		}
+	}
+
+	private void deleteGermplasmListData(final List<GermplasmListData> germplasmListData) {
+		try {
+			for (final GermplasmListData data : germplasmListData) {
+				this.daoFactory.getGermplasmListDataDAO().makeTransient(data);
+			}
+		} catch (final Exception e) {
+			throw new MiddlewareQueryException(
+				"Error encountered while deleting Germplasm List Data: GermplasmServiceImpl.deleteGermplasmListData(germplasmListData="
```

Comment:
Change log message to:  `GermplasmListServiceImpl`. Also correct message in `addGermplasmListData`method